packages/core/src/utils.ts

```diff
@@ -76,15 +77,13 @@ export function resolveId(input: Reference | Resource | undefined): string | und
  * @param reference - A reference to a FHIR resource.
  * @returns A tuple containing the `ResourceType` and the ID of the resource or `undefined` when `undefined` or an invalid reference is passed.
  */
-export function parseReference(reference: Reference): [ResourceType, string] | undefined;
-export function parseReference(reference: undefined): undefined;
-export function parseReference(reference: Reference | undefined): [ResourceType, string] | undefined {
+export function parseReference(reference: Reference | undefined): [ResourceType, string] {
   if (reference?.reference === undefined) {
-    return undefined;
+    throw new OperationOutcomeError(validationError('Reference missing reference property.'));
   }
   const [type, id] = reference.reference.split('/');
   if (type === '' || id === '' || id === undefined) {
-    return undefined;
+    throw new OperationOutcomeError(validationError('Unable to parse reference string.'));
```

Comment:
I'm not sure anyone's actually using this function so I thought it would be okay to make this breaking change. I did a check through the codebase and no one is using it since I added it. 

It's also pretty much useless with the old signature since since the return value can be `undefined` if you destructure the return, it gives a TS error at compile time and a runtime error if we ever return `undefined` in the case of an invalid reference.